backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/StopVmCommandBase.java

```diff
@@ -174,6 +178,7 @@ protected void executeVmCommand() {
 
         suspendedVm = getVm().getStatus() == VMStatus.Suspended;
         if (suspendedVm) {
+            resourceManager.internalSetVmStatus(getVm().getDynamicData(), VMStatus.Down);
```

Comment:
Doesn't this do more than clearing the dynamic pinning data?  Is it safe to call it here?

Reply:
I think it is safe but I think we need @ahadas to have a look to be sure

Reply:
probably needs a check in vms monitoring to see what it will trigger.
i think the safest approach would be just "reset" what we need (the pinning, right?) and somehow catch it in order to set.

Reply:
we do set `getVm().setStatus(VMStatus.Down);` right after that, in the `endVmCommand`, so I think it should be ok.

Reply:
> we do set `getVm().setStatus(VMStatus.Down);` right after that anyway, in the `endVmCommand`, so I think it should be ok.

yeah, so why not to replace that call with `resourceManager.internalSetVmStatus(getVm().getDynamicData(), VMStatus.Down);` ?

Reply:
I tried to change the workflow only for the `suspend` path, not for the others. For other workflows the `resourceManager.internalSetVmStatus` is called in other places.

Reply:
@ljelinkova that's ok, `envVmCommand` is called only when stopping a suspended vm :) we don't have async tasks in other flows so we never get to the end-action phase (the monitoring should detect the vm as down and update the database accordingly). previously, when stopping a suspended vm we started by removing its memory volumes and then update the vm at the end-action phase. [that has changed long time ago](https://github.com/oVirt/ovirt-engine/commit/f2d80ef74584275496d0e9f048f091077f07aba4) - now we update the database and then remove the memory volumes without waiting for remove-image to finish ("fire and forget").
long story short, we can even inline `endVmCommand` here - that's the only place it is called from